### ISSUE TITLE
Bigquery dialect fails to parse setting of collumn descriptions with SET OPTIONS

### ISSUE THREAD

**Issue by mmozerwayfair** [sqlfluff/sqlfluff#4309]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have bigquery query which creates a new table and sets column descriptions following googles documentation here:
[https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#alter_column_set_options_statement](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#alter_column_set_options_statement)
The statement 
```
ALTER COLUMN IF EXISTS column_name SET OPTIONS (
        description='some description here'
    )
```
Produces the error:
```
Line 439, Position 18: Found unparsable section: 'IF
                       | EXISTS column_name SET OPTIONS (...'
```
and then again in the summary
```
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?                                                                                                        
WARNING    Zero length Lex item returned from '<unlexable>'. Report this as a bug. 
```

### Expected Behaviour

The query runs correctly in bigquery, so I'd expect this construct to be parsed by SQLFluff

### Observed Behaviour

The setting of a column description is flagged as "not parsable".
See below for an example. When executed in the bigquery console, the column description is correctly set to "description"

### How to reproduce

CREATE TEMPORARY TABLE test AS (
  SELECT "test" as col
)
;
ALTER TABLE test
ALTER COLUMN col SET OPTIONS(description='description')
;

### Dialect

bigquery

### Version

sqlfluff: 1.4.5
python: 3.8.15

### Configuration

The above example fails on en empty configuration

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by mmozerwayfair** [comment]
Thanks a lot!
I really appreciate!
